src/mono/mono/mini/mini-amd64.c

```diff
@@ -7607,6 +7607,23 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 		case OP_SSSE3_SHUFFLE:
 			amd64_sse_pshufb_reg_reg (code, ins->dreg, ins->sreg2);
 			break;
+		case OP_VECTOR_IABS:
+			switch (ins->inst_c1) {
+			case MONO_TYPE_I1:
+				amd64_sse_pabsb_reg_reg(code, ins->dreg, ins->sreg1);
+				break;
+			case MONO_TYPE_I2:
+				amd64_sse_pabsw_reg_reg(code, ins->dreg, ins->sreg1);
+				break;
+			case MONO_TYPE_I4:
+			case MONO_TYPE_I:
```

Comment:
I *think* that `MONO_TYPE_I` can change size depending on pointer size (32/64-bit). To make sure that it has the same size as `MONO_TYPE_I4`, I would use `#if TARGET_SIZEOF_VOID_P == 4` around it.

Otherwise, it looks good to me, good job!

Reply:
Already had check for that in simd-intrinsics.c. Added one here as well